In [45]:
import requests
import pandas
import json
import os
from pathlib import Path


In [46]:
#Test endpoint to get a sample of dinosaur occurrence data from the Paleobiology Database API
occurrence_url = "https://paleobiodb.org/data1.2/occs/list.json?base_name=Dinosauria&show=full&limit=10"

response = requests.get(occurrence_url)
response.raise_for_status()

occurrence_data = response.json()

occurrence_df = pandas.DataFrame(occurrence_data["records"])

occurrence_df.head()

,oid,cid,idn,tna,att,rnk,tid,oei,eag,lag,...,tps,tpt,tpc,eid,srs,srb,sru,sro,cpa,ccu
0,occ:41524,col:3257,Aves indet.,Aves,(Linnaeus 1758),17,txn:36616,Lutetian,48.07,41.03,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,occ:41580,col:3256,Aves indet.,Aves,(Linnaeus 1758),17,txn:36616,Ypresian,56.00,48.07,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,occ:130209,col:10755,Chaoyangosaurus liaosiensis n. gen. n. sp.,Chaoyangsaurus youngi,Zhao et al. 1999,3,txn:55580,Late Kimmeridgian,152.21,132.60,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,occ:130294,col:10764,Protarchaeopteryx robusta n. gen. n. sp.,Protarchaeopteryx robusta,Ji and Ji 1997,3,txn:66068,Late Barremian,125.77,119.57,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,occ:130295,col:10764,Caudipteryx zoui n. gen. n. sp.,Caudipteryx zoui,Ji et al. 1998,3,txn:66066,Late Barremian,125.77,119.57,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [47]:
#Retrieve the full dataset of dinosaur occurrences from the Paleobiology Database API
full_url = "https://paleobiodb.org/data1.2/occs/list.json?base_name=Dinosauria&show=full&limit=50000"
response = requests.get(full_url)
response.raise_for_status()

full_data = response.json()
full_df = pandas.DataFrame(full_data['records'])
print(full_df.shape)


(37816, 131)


In [48]:
#Check for duplicates in the 'oid' column
print(full_df['oid'].nunique())
print(full_df['oid'].duplicated().sum())

37816
0


In [49]:
#Examine taxonomic scope
print(full_df['tna'].nunique())
print(full_df['rnk'].value_counts())
full_df['tna'].value_counts().head(20)


6173
rnk
3     17316
25     6760
5      6108
9      4554
12      737
8       715
17      632
13      309
11      238
10      196
2       147
16       62
7        30
14       10
4         2
Name: count, dtype: int64


tna
Theropoda          1992
Dinosauria         1422
Sauropoda          1042
Hadrosauridae      1022
Aves                632
Ornithopoda         520
Ceratopsidae        287
Dromaeosauridae     265
Titanosauria        248
Tyrannosauridae     245
Ankylosauria        240
Ornithischia        199
Grallator           195
Passeriformes       167
Coelurosauria       164
Ornithomimidae      159
Titanosauridae      156
Ankylosauridae      149
Ceratopsia          148
Abelisauridae       139
Name: count, dtype: int64

In [50]:
#Save as a CSV in data/raw folder
raw_path = Path("..")/"data"/"raw"
full_df.to_csv(raw_path / "raw_dinosaur_occurrences.csv", index=False)
print((raw_path / "raw_dinosaur_occurrences.csv").exists())

True


Source: Paleobiology Database

Endpoint: https://paleobiodb.org/data1.2/occs/list.json

Query:
base_name=Dinosauria
show=full
limit=50000

Records Retrieved:
37,816

Columns Retrieved: 
131

Download Date:
8/10/2016